File: src/main/kotlin/solutionsJava/NumberSpiral.java
import java.util.Scanner;

public class NumberSpiral {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = scanner.nextInt();
        scanner.nextLine();

        StringBuilder res = new StringBuilder();
        for (int i = 0; i < t; i++) {
            String[] test = scanner.nextLine().trim().split(" ");
            long row = Long.parseLong(test[0]);
            long col = Long.parseLong(test[1]);

            long max = Math.max(row, col);

            long answer;
            if (max % 2 == 1L) {
                if (row == col) {
                    answer = (row * row) - (row - 1);
                } else if (row > col) {
                    answer = ((row - 1) * (row - 1)) + col;
                } else {
                    answer = (col * col) - (row - 1);
                }
            } else {
                if (row == col) {
                    answer = (row * row) - (row - 1);
                } else if (row > col) {
                    answer = (row * row) - (col - 1);
                } else {
                    answer = ((col - 1) * (col - 1)) + row;
                }
            }

            if (i > 0) res.append("\n");
            res.append(answer);
        }

        System.out.println(res);
    }
}

File: src/main/kotlin/solutions/NumberSpiral.kt
package uz.safix.solutions

import java.util.Scanner

fun main() {
    val scanner = Scanner(System.`in`)
    val t = scanner.nextInt(); scanner.nextLine()

    val res = mutableListOf<Long>()
    repeat(t) {
        val test = scanner.nextLine().trim().split(" ")
        val row = test[0].toLong(); val col = test[1].toLong()

        val max = maxOf(row, col)

        if (max % 2 == 1L) {
            if (row == col) {
                res.add((row * row) - (row - 1))
            } else if (row > col) {
                res.add(((row - 1) * (row - 1)) + col)
            } else {
                res.add((col * col) - (row - 1))
            }
        } else {
            if (row == col) {
                res.add((row * row) - (row - 1))
            } else if (row > col) {
                res.add((row * row) - (col - 1))
            } else {
                res.add(((col - 1) * (col - 1)) + row)
            }
        }
    }

    println(res.joinToString(separator = "\n"))
}